File: tkpdcore/src/main/java/com/tokopedia/core/drawer2/data/mapper/NotificationMapper.java
package com.tokopedia.core.drawer2.data.mapper;

import com.tokopedia.core.drawer2.data.pojo.notification.NotificationData;
import com.tokopedia.core.drawer2.data.pojo.notification.NotificationModel;
import com.tokopedia.core.network.ErrorMessageException;
import com.tokopedia.core.network.retrofit.response.TkpdResponse;

import javax.inject.Inject;

import retrofit2.Response;
import rx.functions.Func1;

/**
 * Created by nisie on 5/5/17.
 */

public class NotificationMapper implements Func1<Response<TkpdResponse>, NotificationModel> {

    @Inject
    public NotificationMapper() {
    }

    @Override
    public NotificationModel call(Response<TkpdResponse> response) {
        NotificationModel model = new NotificationModel();

        if (response.isSuccessful()) {
            if (!response.body().isError()) {
                NotificationData data = response.body().convertDataObj(NotificationData.class);
                model.setSuccess(true);
                model.setNotificationData(data);
            } else {
                if (response.body().getErrorMessages() == null
                        && response.body().getErrorMessages().isEmpty()) {
                    model.setSuccess(false);
                } else {
                    throw new ErrorMessageException(response.body().getErrorMessageJoined());
                }
            }
            model.setStatusMessage(response.body().getStatusMessageJoined());
            model.setResponseCode(response.code());
        } else {
            throw new RuntimeException(String.valueOf(response.code()));
        }
        return model;
    }
}

File: features/merchant/seller_home_drawer/src/main/java/com/tokopedia/sellerhomedrawer/domain/mapper/NotificationMapper.kt
package com.tokopedia.sellerhomedrawer.domain.mapper

import com.tokopedia.abstraction.common.network.response.TokopediaWsV4Response
import com.tokopedia.network.exception.MessageErrorException
import com.tokopedia.sellerhomedrawer.data.drawernotification.NotificationData
import com.tokopedia.sellerhomedrawer.data.drawernotification.NotificationModel
import retrofit2.Response
import rx.functions.Func1
import javax.inject.Inject

class NotificationMapper @Inject constructor() : Func1<Response<TokopediaWsV4Response>, NotificationModel> {

    override fun call(response: Response<TokopediaWsV4Response>?): NotificationModel {
        val model = NotificationModel()
        if (response != null) {
            if (response.isSuccessful) {
                val isError = response.body()?.isError
                if (isError != null && !isError) {
                    val data = response.body()?.convertDataObj(NotificationData::class.java)
                    model.isSuccess = true
                    model.notificationData = data
                } else {
                    val error = response.body()?.errorMessages
                    if (error.isNullOrEmpty())
                        model.isSuccess = false
                    else throw MessageErrorException(response.body()?.errorMessageJoined)
                }
            } else {
                throw RuntimeException("${response.code()}")
            }
        }
        return model
    }
}